_drafts/autolab-and-docker.markdown

```diff
@@ -0,0 +1,56 @@
+---
+layout: post
+title: "Autolab and Docker"
+author: Mihir Pandya
+
+tags: [autograding, docker, tango]
+---
+
+![]({{site-url}}/assets/docker.svg)
+
+As highlighted in our ['Hello World' post][hello-world], we at Autolab want to make Autograding for all a reality. In pursuit of this vision, we started by making Autolab robust enough to be usable by a variety of classes. Following that, we [open-sourced][gh-autolab] this project after it had been tried and tested at Carnegie Mellon University. 
+
+While these events have been instrumental in making Autolab usable, the time and effort that needs to be spent in setting up and configuring Tango, the autograder, is non-trivial. Furthermore, Tango supports a limited number of platforms (Local, EC2) on which it can run jobs. Hence, as part of the latest efforts in making Autograding for all a reality, we are [integrating Docker][gh-docker] with Autolab in order to significantly reduce the overhead in getting started and to make autograding platform-agnostic. 
```

Comment:
- configuring Tango, the __autograding backend__,
  - the instructors write the autograders
- Furthermore, Tango __needs an environment in which it can run jobs. It had supported a CMU-specific environment (called Tashi), as well as supporting running jobs locally on a single server or laptop, or even a cluster of EC2 machines. However, moving forward we wanted to provide more options that simplified the process of developing, trialling, and deploying Tango.__ Hence, as ...